docs/contracts/accounting-oracle.md

```diff
@@ -21,3 +25,511 @@ This would ultimately result in no oracle reports and no stETH rebases for this
+```
+
+### MANAGE_CONSENSUS_CONTRACT_ROLE()
+An ACL role granting the permission to set the consensus contract address by calling setConsensusContract.
+```solidity
+bytes32 public constant MANAGE_CONSENSUS_CONTRACT_ROLE = keccak256("MANAGE_CONSENSUS_CONTRACT_ROLE");
+```
+
+### MANAGE_CONSENSUS_VERSION_ROLE()
+An ACL role granting the permission to set the consensus version by calling setConsensusVersion.
+```solidity
+bytes32 public constant MANAGE_CONSENSUS_VERSION_ROLE = keccak256("MANAGE_CONSENSUS_VERSION_ROLE");
+```
+
+## Events
+
+### ExtraDataSubmitted()
+Emits when any extra report data for the current reporting frame has been submitted.
+```solidity
+ExtraDataSubmitted(uint256 indexed refSlot, uint256 itemsProcessed, uint256 itemsCount)
+```
+
+### WarnExtraDataIncompleteProcessing()
+Emits when try to submit the same report, but not all items are processed yet.
+```solidity
+event WarnExtraDataIncompleteProcessing(
+    uint256 indexed refSlot,
+    uint256 processedItemsCount,
+    uint256 itemsCount
+)
+```
+
+### ConsensusHashContractSet()
+Emits when a contract hash value is changed.
+```solidity
+event ConsensusHashContractSet(address indexed addr, address indexed prevAddr)
+```
+
+### ConsensusVersionSet()
+Emits when a consensus version value is changed.
+```solidity
+event ConsensusVersionSet(uint256 indexed version, uint256 indexed prevVersion)
+```
+
+### ReportSubmitted()
+Emits when a new consensus report hash is submitted
+```solidity
+event ReportSubmitted(uint256 indexed refSlot, bytes32 hash, uint256 processingDeadlineTime)
+```
+
+### ReportDiscarded()
+Emits when consensus report is discarded.
+
+```solidity
+event ReportDiscarded(uint256 indexed refSlot, bytes32 hash)
+```
+
+### ProcessingStarted()
+Emits when report data is submitted
+```solidity
+event ProcessingStarted(uint256 indexed refSlot, bytes32 hash)
+```
+
+### WarnProcessingMissed()
+Emits on [submitConsensusReport](#submitconsensusreport) when `refSlot != prevSubmittedRefSlot && prevProcessingRefSlot != prevSubmittedRefSlot`
+```solidity
+event WarnProcessingMissed(uint256 indexed refSlot)
+```
+
+## Reverts
+
+### submitReportData()
+
+To ensure that the reported data is within possible values, the handler function performs a number of sanity checks. When checking, reverts may occur in different contracts. The diagram shows the interaction with other contracts, and below is a description of where, what and why the reversal occurs.
+
+
+```mermaid
+graph LR;
+  A[/  \]--submitReportData-->AccountingOracle--handleConsensusLayerReport--->LegacyOracle;
+  AccountingOracle--handleOracleReport-->Lido--handlePostTokenRebase-->LegacyOracle
+  AccountingOracle--checkAccountingExtraDataListItemsCount-->OracleReportSanityChecker;
+  AccountingOracle--updateExitedValidatorsCountByStakingModule-->StakingRouter;
+  AccountingOracle--checkExitedValidatorsRatePerDay-->OracleReportSanityChecker;
```

Comment:
Looks like `WithdrawalQueue.onOracleReport()` is missing here

Reply:
wow, the diagram is impressive :star_struck: